test/support/async.ex

```diff
@@ -55,6 +62,28 @@ defmodule Support.Async do
     end
   end
 
+  defmacro async_setup(context, block) do
+    do_setup(context, block)
+  end
+
+  defmacro async_setup(block) do
+    do_setup(quote(do: %{}), block)
+  end
+
+  defp do_setup(context, block) do
+    name = :"async_setup_#{:erlang.unique_integer([:positive])}"
+
+    quote bind_quoted: [context: escape(context), block: escape(block), name: name] do
```

Comment:
I guess a pattern match on the context won't work this way (like `async_setup %{key: value} do ... end`)